modules/openapi-generator/src/main/resources/dart-dio/api.mustache

```diff
@@ -23,14 +23,18 @@ class {{classname}} {
         {{{dataType}}} {{paramName}},{{#-last}} { {{/-last}}{{/requiredParams}}{{#optionalParams}}
         {{{dataType}}} {{paramName}},{{/optionalParams}}
         CancelToken cancelToken,
-        Map<String, String> headers,
+        Map<String, dynamic> headers,
+        Map<String, dynamic> extra,
+        ValidateStatus validateStatus,
         ProgressCallback onSendProgress,
         ProgressCallback onReceiveProgress,
     }) async {
         final String _path = '{{{path}}}'{{#pathParams}}.replaceAll('{' r'{{baseName}}' '}', {{{paramName}}}.toString()){{/pathParams}};
 
         final Map<String, dynamic> queryParams = {};
-        final Map<String, String> headerParams = Map.from(headers ?? {});
+        final Map<String, dynamic> headerParams = {
+            if (headers != null) ...headers,
```

Comment:
never seen this syntax before. basically `headerParams` gets content of `headers` if `headers` is initialised?

Reply:
Ya basically like the ecma spread operator. Works for all collections.